### ISSUE TITLE
PropertyGrid and SQLConnectionStringBuilder

### ISSUE THREAD

**Issue by FJGR65** [dotnet/SqlClient#1142]
When I try to edit the properties of the SqlConnectionStringBuilder in a PropertyGrid there is a problem with the synonyms. Properties appear with their original names and synonyms.
Categories don't work either, as do descriptions.

The application is on .NET 5.0 

![Portapapeles01](https://user-images.githubusercontent.com/73061777/123539341-8f2b0100-d739-11eb-8f05-4b1702e88f12.jpg)

**Comment by Wraith2** [comment]
Runtime or design time? The ref assembly looks like it's correctly annotated with ComponentModel attributes so it should behave properly at design time. The implementation in the src assembly doesn't have the attributes though, so if this is runtime then we can fix it, if it's design time then I'm not sure what's happening.

**Comment by FJGR65** [comment]
It's at runtime.

Thank you for the correction.

**Comment by Wraith2** [comment]
It should be pretty simple to fix then. We just need to annotate the src with the same attributes as ref.

**Comment by JRahnama** [comment]
@FJGR65 Thanks for bringing  up this issue to our attention. We will look into it shortly.
@Wraith2 Thank you for the solution.

**Comment by JRahnama** [comment]
@FJGR65 I added the annotations to src, but when it comes to testing I had some difficulties creating the grid. Can you walk me through the steps you have taken to make the property grid in netcore please?

**Comment by FJGR65** [comment]
Sorry. I'm in holidays.
Tomorrow send te code.

**Comment by FJGR65** [comment]
Good Morning:

I do not know how to upload a project to the comments, so I send you the code by e-mail. It is a reduced version of what I am doing, but I hope it is useful to test your annotations.

**Comment by JRahnama** [comment]
> Good Morning:
> 
> I do not know how to upload a project to the comments, so I send you the code by e-mail. It is a reduced version of what I am doing, but I hope it is useful to test your annotations.

Thank you. I got the code. I will start testing the fixes and will update you shortly.

**Comment by FJGR65** [comment]
Thanks. I'm at the beach, but if you need anything else, I can send it to
you tomorrow.

**Comment by JRahnama** [comment]
@FJGR65 I tested and everything is working as expected. Please check the fixe in your convenient time and confirm that the issue is solved.

Thanks for bringing up the issue.
Regards

**Comment by FJGR65** [comment]
Thank you very much for the solution. There are no problems with the synonyms, although the categories and descriptions continue without appearing. However, for me it is enough.

**Comment by ErikEJ** [comment]
@FJGR65 Can you share a screenshot that demonstrates the missing parts?

**Comment by FJGR65** [comment]
This is with new fixes:
![Portapapeles02](https://user-images.githubusercontent.com/73061777/124759496-f9a92180-df2f-11eb-83c5-b7c255045b7a.jpg)

This is with .Net Framework 4.5.2 with System.Data.SqlClient (in spanish, sorry)

![Portapapeles03](https://user-images.githubusercontent.com/73061777/124759617-1e04fe00-df30-11eb-93d9-2edd49fe2fca.jpg)

**Comment by Wraith2** [comment]
Missing CategoryAttribute annotations.

**Comment by FJGR65** [comment]
And descriptions not show.

**Comment by Wraith2** [comment]
We have localized versions of CategoryAttribute and DescriptionAttribute called ResCategoryAttribute and ResDescriptionAttribute which allow localization, those will need to be applied.

**Comment by cheenamalhotra** [comment]
CategoryAttributes could be applied to match the behavior with .NET Framework, but we don't have localized resources suported in .NET Core yet.

That would be part of a separate task, just FYI. cc @David-Engel 
If it can work in English, it should be acceptable fix for this particular issue.

**Comment by Wraith2** [comment]
I think english would be better than nothing, at least there's an indication of category in that case even if the category itself isn't in the right language.

**Comment by JRahnama** [comment]
> That would be part of a separate task, just FYI. cc @David-Engel David Engel (Simba Technologies Inc) Vendor
> If it can work in English, it should be acceptable fix for this particular issue.

David approved that English-only is better than nothing.

There are several changes made to the PR. More file has changed.

@FJGR65 can you check with the new build from the PR?

**Comment by FJGR65** [comment]
First, thank all of you for the effort made to solve the problem. The result has been much more than satisfactory, since my intention was simply to solve the problem with the PropertyGrid.
You have managed to work properly, to show the categories and descriptions. It is much more than expected. Thank you.

As for the issue of translations, I just hoped they were in English. The screen shown in Spanish was because I am on holidays and it is very difficult for me to access my development computer on remote, so I could not show the screen in English. It has also been rather complicated to respond more quickly to your requirements, since I do not have access to the Internet on the beach and I have had to move to perform the tests and screenshots.

Anyway, it has been very rewarding to check the quick response from you, kindness and willingness to solve the problem. Thanks.

Finally, I show you the last screenshot in which it is checked that the PropertyGrid works correctly. Thus, for me, the problem is completely solved.

![Portapapeles04](https://user-images.githubusercontent.com/73061777/124898106-2d8f5000-dfdf-11eb-96b0-806025010ed6.jpg)


All the best
Francisco

**Comment by ErikEJ** [comment]
So Categories work on an English system?

**Comment by FJGR65** [comment]
Is working in a spanish system perfectly. For me its ok.

**Comment by JRahnama** [comment]
> So Categories work on an English system?

Descriptions are English-only for now.

**Comment by Wraith2** [comment]
> but we don't have localized resources supported in .NET Core yet.

If there isn't an issue to track localization in netcore can we open one? The localized resources are in netfx so in theory it's all the waiting on the source merges.
